src/contacts/contacts.py

```diff
@@ -118,17 +120,22 @@ def add_contact(username):
                 "is_external": req["is_external"],
             }
             # Add contact_data to database
+            app.logger.debug("Adding new contact to the database.")
             contacts_db.add_contact(contact_data)
+            app.logger.info("Successfully added new contact.")
             return jsonify({}), 201
 
-        except (PermissionError, jwt.exceptions.InvalidTokenError):
+        except (PermissionError, jwt.exceptions.InvalidTokenError) as err:
+            app.logger.error("Error adding contact: %s", str(err))
             return "authentication denied", 401
         except UserWarning as warn:
+            app.logger.error("Error adding contact: %s", str(warn))
```

Comment:
This logging level should not be `error`

This situation is invalid user input.  The application is operating normally when it throws this exception.

Instead it should be either `warn` (if you believe users are always perfect at following rules and this is unexpected behavior), or it should be `info` (because it would be nice to know that the contact could not be added).  Again, you can update the log message to reflect this:  e.g. "Failed to add contact:  invalid input"